appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -140,12 +140,18 @@ internal class ExperienceRenderer(
         renderContextManager.getStateMachine(renderContext)?.handleAction(StartStep(stepReference))
     }
 
-    suspend fun preview(experienceId: String): Boolean {
+    data class PreviewError(val experience: Experience)
+
+    suspend fun preview(experienceId: String): PreviewError? {
         repository.getExperiencePreview(experienceId)?.let {
-            return show(it)
+            renderContextManager.putPreview(it)
+
+            if (show(it)) return null
+
+            return PreviewError(it)
```

Comment:
If this is a preview of a tooltip that is not found, that would also give an error on show, correct? Could we have a situation where the "Please navigate to screen with frame..." error is invalid/confusing? Wondering if we need more targeted error handling specific to embeds and frame not found cases?

Reply:
FYI for iOS, I've added the embed error, but also handling for network errors and a fallback to a generic message: https://github.com/appcues/appcues-ios-sdk/blob/b96359ca9c02e6f590bbf2cc244a9c0ccd448f9d/Sources/AppcuesKit/Presentation/DeepLinkHandler.swift#L134-L153